lib/resources/aws/aws_ec2_instances.rb

```diff
@@ -0,0 +1,72 @@
+# author: Christoph Hartmann
+class AwsEc2Instances < Inspec.resource(1)
+  name 'aws_ec2_instances'
+  desc 'Verifies settings for an EC2 instance'
+  supports platform: 'aws'
+  example "
+    describe_ec2_instances.where { public_ip_address !=nil } do
+      its('instance_id') { should eq [] }
+    end
+  "
+
+  # TODO: rewrite to avoid direct injection, using AwsPluralResourceMixin
+  def initialize(opts = {}, conn = nil)
+    @opts = opts
+    @ec2_client = conn ? conn.ec2_client : inspec_runner.backend.aws_client(Aws::EC2::Client)
+  end
+
+  # Set up some accessors and the filter table
+  filter = FilterTable.create
+  filter.add_accessor(:where).add_accessor(:entries)
```

Comment:
Ahh, I see why you're not using `exist` - it's not there!  You should add it.  It's simple, 
`filter.add(:exists?) { |x| !x.entries.empty? }`

That's pretty much standard in all of our plural resources.  Well, [pretty much](#2779).

Reply:
I also avoid it because I want an explicit list of failed instances, rather than a `failed` message.